src/D2L.CodeStyle.Analyzers/Immutability/ImmutableDefinitionChecker.cs

```diff
@@ -333,6 +333,18 @@ ExpressionSyntax expression
 					assignedType: assignedType
 				);
 			}
+
+			public static AssignmentInfo Create(

```

Comment:
Could the constructor just be public?

Reply:
Yeah for sho. Discovering the end solution organically.